### PR TITLE
Replaces checkpass with bitfield checks

### PR DESCRIPTION
**Pull request by DominikPanic**
/tg/ and good code bones

### CODE REVIEW

File: code/game/atoms.dm [hunk 1]
```diff
@@ -369,8 +369,6 @@ its easier to just keep the beam vertical.
 	else
 		return 0
```
Comment (psykzz) [inline code/game/atoms.dm]:
```suggestion
		return FALSE
```

File: code/game/machinery/OpTable.dm [hunk 2]
```diff
@@ -130,7 +130,7 @@
 
 
 /obj/machinery/optable/CanPass(atom/movable/mover, turf/target)
-	if(istype(mover) && mover.checkpass(PASSTABLE))
+	if(istype(mover) && CHECK_BITFIELD(mover.flags_pass, PASSTABLE))
 		return 1
```
Comment (psykzz) [inline code/game/machinery/OpTable.dm]:
```suggestion
		return TRUE
```

File: code/game/machinery/OpTable.dm [hunk 3]
```diff
@@ -130,7 +130,7 @@
...
 	else
 		return 0
```
Comment (psykzz) [inline code/game/machinery/OpTable.dm]:
```suggestion
		return FALSE
```

File: code/game/machinery/deployable.dm [hunk 4]
```diff
@@ -106,7 +106,7 @@
 	CanPass(atom/movable/mover, turf/target, height = 0, air_group = 0)//So bullets will fly over and stuff.
 		if(air_group || (height == 0))
 			return 1
```
Comment (psykzz) [inline code/game/machinery/deployable.dm]:
```suggestion
			return TRUE
```

File: code/game/machinery/deployable.dm [hunk 5]
```diff
@@ -106,7 +106,7 @@
...
-		if(istype(mover) && mover.checkpass(PASSTABLE))
+		if(istype(mover) && CHECK_BITFIELD(mover.flags_pass, PASSTABLE))
 			return 1
```
Comment (psykzz) [inline code/game/machinery/deployable.dm]:
```suggestion
			return TRUE
```

File: code/game/machinery/deployable.dm [hunk 6]
```diff
@@ -106,7 +106,7 @@
...
 		else
 			return 0
```
Comment (psykzz) [inline code/game/machinery/deployable.dm]:
```suggestion
			return FALSE
```

File: code/game/machinery/spaceheater.dm [hunk 7]
```diff
@@ -27,7 +27,7 @@
 /obj/machinery/space_heater/CanPass(atom/movable/mover, turf/target)
 	if(!density) //Because broken racks -Agouri |TODO: SPRITE!|
 		return 1
```
Comment (psykzz) [inline code/game/machinery/spaceheater.dm]:
```suggestion
		return TRUE
```

File: code/game/machinery/spaceheater.dm [hunk 8]
```diff
@@ -27,7 +27,7 @@
...
-	if(istype(mover) && mover.checkpass(PASSTABLE))
+	if(istype(mover) && CHECK_BITFIELD(mover.flags_pass, PASSTABLE))
 		return 1
```
Comment (psykzz) [inline code/game/machinery/spaceheater.dm]:
```suggestion
		return TRUE
```

File: code/game/machinery/spaceheater.dm [hunk 9]
```diff
@@ -27,7 +27,7 @@
...
 	else
 		return 0
```
Comment (psykzz) [inline code/game/machinery/spaceheater.dm]:
```suggestion
		return FALSE
```

File: code/game/objects/structures/crates_lockers/crates.dm [hunk 10]
```diff
@@ -22,7 +22,7 @@
 	return 1
```
Comment (psykzz) [inline code/game/objects/structures/crates_lockers/crates.dm]:
```suggestion
	return TRUE
```

File: code/game/objects/structures/crates_lockers/crates.dm [hunk 11]
```diff
@@ -22,7 +22,7 @@
...
 
 /obj/structure/closet/crate/CanPass(atom/movable/mover, turf/target)
-	if(istype(mover) && mover.checkpass(PASSTABLE))
+	if(istype(mover) && CHECK_BITFIELD(mover.flags_pass, PASSTABLE))
 		return 1
```
Comment (psykzz) [inline code/game/objects/structures/crates_lockers/crates.dm]:
```suggestion
		return TRUE
```

File: code/game/objects/structures/reagent_dispensers.dm [hunk 12]
```diff
@@ -51,7 +51,7 @@
 	return
 
 /obj/structure/reagent_dispensers/CanPass(atom/movable/mover, turf/target)
-	if(istype(mover) && mover.checkpass(PASSTABLE))
+	if(istype(mover) && CHECK_BITFIELD(mover.flags_pass, PASSTABLE))
 		return 1
```
Comment (psykzz) [inline code/game/objects/structures/reagent_dispensers.dm]:
```suggestion
		return TRUE
```

File: code/game/objects/structures/windoor_assembly.dm [hunk 13]
```diff
@@ -49,15 +49,15 @@ obj/structure/windoor_assembly/Destroy()
 	icon_state = "[facing]_[secure]windoor_assembly[state]"
 
 /obj/structure/windoor_assembly/CanPass(atom/movable/mover, turf/target)
-	if(istype(mover) && mover.checkpass(PASSGLASS))
+	if(istype(mover) && CHECK_BITFIELD(mover.flags_pass, PASSGLASS))
 		return 1
```
Comment (psykzz) [inline code/game/objects/structures/windoor_assembly.dm]:
```suggestion
		return TRUE
```

File: code/game/objects/structures/windoor_assembly.dm [hunk 14]
```diff
@@ -49,15 +49,15 @@ obj/structure/windoor_assembly/Destroy()
...
 	if(get_dir(loc, target) == dir) //Make sure looking at appropriate border
 		return !density
 	else
 		return 1
```
Comment (psykzz) [inline code/game/objects/structures/windoor_assembly.dm]:
```suggestion
		return TRUE
```

File: code/game/objects/structures/windoor_assembly.dm [hunk 15]
```diff
@@ -49,15 +49,15 @@ obj/structure/windoor_assembly/Destroy()
...
 
 /obj/structure/windoor_assembly/CheckExit(atom/movable/mover as mob|obj, turf/target as turf)
-	if(istype(mover) && mover.checkpass(PASSGLASS))
+	if(istype(mover) && CHECK_BITFIELD(mover.flags_pass, PASSGLASS))
 		return 1
```
Comment (psykzz) [inline code/game/objects/structures/windoor_assembly.dm]:
```suggestion
		return TRUE
```

File: code/modules/hydroponics/hydro_tray.dm [hunk 16]
```diff
@@ -142,7 +142,7 @@
 
 /obj/machinery/portable_atmospherics/hydroponics/CanPass(atom/movable/mover, turf/target)
 	if(!density) return 1
```
Comment (psykzz) [inline code/modules/hydroponics/hydro_tray.dm]:
```suggestion
	if(!density)
		return TRUE
```

File: code/modules/hydroponics/hydro_tray.dm [hunk 17]
```diff
@@ -142,7 +142,7 @@
...
-	if(istype(mover) && mover.checkpass(PASSTABLE))
+	if(istype(mover) && CHECK_BITFIELD(mover.flags_pass, PASSTABLE))
 		return 1
```
Comment (psykzz) [inline code/modules/hydroponics/hydro_tray.dm]:
```suggestion
		return TRUE
```

File: code/modules/hydroponics/hydro_tray.dm [hunk 18]
```diff
@@ -142,7 +142,7 @@
...
 	else
 		return 0
```
Comment (psykzz) [inline code/modules/hydroponics/hydro_tray.dm]:
```suggestion
		return FALSE
```

**Comment by DominikPanic** [comment]
@psykzz I specifically avoided doing that due to the scope.

**Comment by Rohesie** [comment]
For code that's unchanged by the PR we avoid asking for such changes, @psykzz .

**Comment by psykzz** [comment]
@DominikPanic @Rohesie, no problem just checking here.

**Pull request merged**